tests/TestCase/ORM/Association/HasManyTest.php

```diff
@@ -157,8 +158,55 @@ public function testSetSort(): void
     {
         $assoc = new HasMany('Test');
         $this->assertNull($assoc->getSort());
+
+        $assoc->setSort('id ASC');
+        $this->assertSame('id ASC', $assoc->getSort());
+
         $assoc->setSort(['id' => 'ASC']);
-        $this->assertEquals(['id' => 'ASC'], $assoc->getSort());
+        $this->assertSame(['id' => 'ASC'], $assoc->getSort());
+
+        $closure = function () {
+            return ['id' => 'ASC'];
+        };
+        $assoc->setSort($closure);
+        $this->assertSame($closure, $assoc->getSort());
+
+        $expression = new OrderClauseExpression('id', 'ASC');
+        $assoc->setSort($expression);
+        $this->assertSame($expression, $assoc->getSort());
+    }
+
+    /**
+     * Tests that sorting works using the accepted types for `setSort()`.
+     */
+    public function testSorting(): void
+    {
+        $authors = $this->getTableLocator()->get('Authors');
+        $assoc = $authors->hasMany('Articles');
+
+        $field = 'Articles.id';
+        $driver = $authors->getConnection()->getDriver();
+        if ($driver->isAutoQuotingEnabled()) {
+            $field = $driver->quoteIdentifier($field);
+        }
+
+        $assoc->setSort("$field DESC");
+        $result = $authors->get(1, ['contain' => 'Articles']);
+        $this->assertSame([3, 1], array_column($result['articles'], 'id'));
+
+        $assoc->setSort(["$field DESC"]);
```

Comment:
Shouldn't this be a different sort than the same test above?

Reply:
Ouch, yes, copy pasta strikes again.